tests/multiply.apln

```diff
@@ -0,0 +1,154 @@
+:Namespace multiply
+    Assert←#.unittest.Assert
+    stripToSameLen←#.utils.stripToSameLen
+    isDyalogClassic←#.utils.isClassic
+
+    complexMultiplyModel←{
+        (⎕FR≡1287) ∧ case≡'Hcmplx': ⍺×⍵ ⍝ Skipping cmplx for ⎕FR←1287
+        a c b d←∊(9 11○⊂)⍺ ⍵ 
+        ⍝ using formula (a+bi)(c+di)=(ac−bd)+(ad+bc)i
+        x←((a model c)-(b model d))
+        y←((a model d)+(b model c))
+        x(⊣+¯11○⊢)y
+    }
+
+    ⍝ model for multiply might not be completely accurate for floats
+    ⍝ because it uses reciprocal and divide and there might be a
+    ⍝ loss of precision
+    model←{⍺{
+        0≡⍵:0
+        ⍝ is ⍵ or ⍺ complex
+        (1289≡⎕DR ⍵) ∨ (1289≡⎕DR ⍺): ⍺ complexMultiplyModel¨ ⍵
+        ⍺⌹¨÷⍵
+    }¨⍵}
+
+    ∇ {r}←test_multiply;fr_dbl;fr_decf;RunVariations;bool;i1;i2;i3;char0;char1;char2;char3;charptr;dbl;cmplx;Hcmplx;Hdbl;Sdbl;fl;Hfl;testDesc;case;case2;fr;desc;data;d1;d2;data2;case3;data3;c1;c2;f
+        RunVariations←(model #.testfns._RunVariations_ ×)
+
+        ⍝ constants
+        fr_dbl←#.utils.fr_dbl
+        fr_decf←#.utils.fr_decf
+
+        ⍝ All data generated is unique
+        bool←0 1                                      ⍝ 11: 1 bit Boolean type arrays
+        i1←{⍵,-⍵}⍳120                                 ⍝ 83: 8 bits signed integer
+        i2←{⍵,-⍵}10000+⍳100                           ⍝ 163: 16 bits signed integer
+        i3←{⍵,-⍵}100000+⍳100                          ⍝ 323: 32 bits signed  integer
+        char0←⎕AV                                     ⍝ 82: DyalogAPL classic char set
+        :If ~isDyalogClassic
+            char1←⎕UCS (100+⍳100)                     ⍝ 80: 8 bits character
+            char2←⎕UCS (1000+⍳100)                    ⍝ 160: 16 bits character
+            char3←⎕UCS (100000+⍳100)                  ⍝ 320: 32 bits character
+        :EndIf
+        charptr←(13↑⎕a) (13↓⎕a)                       ⍝ 326: Pointer (32-bit or 64-bit as appropriate)
+        dbl←{⍵,-⍵}i3+0.1                              ⍝ 645: 64 bits Floating
+        cmplx←{⍵,-⍵}(0J1×⍳100)+⌽⍳100                  ⍝ 1289: 128 bits Complex
+        Hcmplx←{⍵,-⍵}(1E14J1E14×⍳20)                  ⍝ 1289 but larger numbers to test for CT value
+        ⍝ Hdbl is 645 but larger numbers to test for CT value
+        ⍝ intervals of 2 are chosen because CT for these numbers +1 and -1
+        ⍝ come under the region of tolerant equality
+        Hdbl←{⍵,-⍵}1E14+(2×⍳50)                       ⍝ 645: large numbers

```

Comment:
The more bigger values are supposed to be tested by the random number generator in `RunVariations` but because of the limitations of the current model I had turned it off. I will move to your model and try it out.

Reply:
Is there a way I can easily find these 'interesting values' in mantis? For now, are there any specific values you would want to include other than this?